async-context/README.md

```diff
@@ -0,0 +1,360 @@
+| function x()            |                              |
+---------------------------                              \/                   
+| function myLoggingAPI() |                  Continuation Invocation ci3                        
+===================================                      /\
+|    Continuation f2()            |                      |
+===================================            -----------
+|    ...host code...      |                              |
+---------------------------                              \/
+|    ...host code...      |                   Continuation Invocation ci2                   
+===================================                      /\
+|   Host Continuation B()         |                      |
+===================================            -----------
+```
+
+## Link Context
+We define the `Link Context` `link-c` of a `Continuation` `c` as a pointer from `c` to the `Continuation Invocation` instance where `c` was constructed.  In the example from above, the `continuation` `f2` has a `link context` pointing to `Continuation Invocation` `ci1`.
+
+## Ready Context
+Consider the following example:
+
+```javascript
+const p = new Promise((resolve, reject) => {
+    setTimeout(function f1() {
+       p.then(function f2() {
+             myLoggingAPI('hello');
+         });
+     }, 1000);
+
+     setTimeout(function f3() {
+         resolve(true);
+     }, 2000)
+});
+```
+
+In this code example, we have:
+  - `setTimeout` and `Promise.then` as our `continuation points`. 
+  - four `continuations` - `f1`, `f2` and `f3`, plus the "root continuation" that 
+    the initial code is executing in. 
+  - four `Continuation Invocations`.
+
+We define the `Ready Context` (//TODO, close on naming - `Causal Context`, `Resolve Context`???) for a promise as a reference from a `Continuation Invocation` associated with a Promise.then call, to the `Continuation Invocation` in which the preceding promise in the promise chain was resolved.
+
+For non-promise-based APIs, the `Ready Context` is the same as the `Link Context` (//TODO or should it be null/undefined?)
+
+In our example above, the `Continuation Invocation` associated with `continuation f2()` has as its `Ready Context`, a reference to the `Continuation Invocation` associated with `continuation f3()`.  
+
+// TODO:  add stacks w/ `continuation invocations` to illustrate `ready context`.
+
+## Crisper Definitions
+
+  - **Continuation Point** - a function that accepts a `Continuation` as a parameter.  This logically represents an async boundary; functions passed to a `Continuation Point` are invoked at some later point in time.
+
+  - **Continuation** -  a function that, when invoked, creates a new `Continuation Invocation` instance, and maintains references to other related `Continuation Invocation` instances. Using TypeScript for descriptive purposes, a `Continuation` has the following shape:
+
+    ```TypeScript
+    interface Continuation {
```

Comment:
Is it something like `Continuation extends Function`?

Reply:
updated.